### PR TITLE
Touchscreen mode for battlefield

### PR DESCRIPTION
**Pull request by Nordsoft91**
Implements https://github.com/vcmi/vcmi/issues/983

Probably not juicy as it described, but
1. It solves problem
2. Better than nothing

Appreciate for testing and feedback

### CODE REVIEW

File: launcher/firstLaunch/firstlaunch_moc.cpp [hunk 1]
```diff
@@ -158,6 +159,19 @@ void FirstLaunchView::exitSetup()
 		mainWindow->exitSetup();
 }
 
+// Initial platform-dependend settings
+void FirstLaunchView::setupPlatformSettings()
```
Comment (IvanSavenko) [inline launcher/firstLaunch/firstlaunch_moc.cpp:163]: Android currently uses separate launcher so this code won't work on Android.

I actually had same issue recently - ended up creating separate options ( "swipe" and "swipeDesktop") for desktop/mobile platforms, with desktop having set it to false by default and on mobile - set to true.

Ugly workaround, sure, but same can be said about this code. Not sure which option is better.

**Comment by IvanSavenko** [comment]
Was planning on doing this myself for 1.3 if nobody else picks it up. And using similar approach as yours.

Better than nothing, sure, although I would've added some UI indication, e.g. semi-transparent unit shadow to show some reaction on 1st click.

File: launcher/firstLaunch/firstlaunch_moc.cpp (hunk 1 above)
Comment (kambala-decapitator) [reply launcher/firstLaunch/firstlaunch_moc.cpp:163]: > Android currently uses separate launcher so this code won't work on Android.

looking forward to solving that for 1.3, already have some initial work done

**Comment by Nordsoft91** [comment]
But it will be reaction at 1st click: as you don't have mouse move on touchscreens, first click means hovering hex with visible intention

**Comment by IvanSavenko** [comment]
> But it will be reaction at 1st click: as you don't have mouse move on touchscreens, first click means hovering hex with visible intention

Well, good enough for first iteration I guess, can be improved later. 

Shadow image would still be much better indication, but I guess not critical for now.

File: launcher/firstLaunch/firstlaunch_moc.cpp (hunk 1 above)
Comment (IvanSavenko) [reply launcher/firstLaunch/firstlaunch_moc.cpp:163]: We would probably still need possibility to defined default settings per platform - we already have cursor, map swiping/panning, and now UI for combat.

Will add this to my 1.3 todo-list, should be simple enough

**Review by IvanSavenko — APPROVED** [review]

**Comment by Nordsoft91** [comment]
tested - works well

**Pull request merged**